# STAT 301: Midterm 1

## Data Wrangling



In [2]:
# The code will not be displayed in the quiz
library(tidyverse, quietly = T)
library(broom, quietly = T)
library(repr, quietly = T)
library(infer, quietly = T)
library(ggplot2)
library(cowplot)
library(ISLR)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘cowplot’


The following object is masked from ‘package:lubridate’:

    stamp




ERROR: Error in library(contextual): there is no package called ‘contextual’



## Question 1

Answer TRUE/FALSE to the following statement:

A hospital is planning to research whether a healthy diet reduces the chance of post-surgery complications in patients who received a kidney transplant. The researchers will monitor the patients for one month after the surgery and register their food intake. A team of nutritionists will classify each patient's diet as: 

**A**: healthy

or

**B**: unhealthy

Is the statement below true or false? 

> Since the hospital is categorizing the patients by their diet, we can conclude that any significant difference in the proportions of patients with and without post-surgery complications is due to the different diets' type: healthy and unhealthy.

### BEGIN SOLUTION

FALSE. The patients are choosing what to eat according to other habits and sociocharacteristics. In this observational study, there may be confounding factors that influence patients' eating choices as well as their recovery. We can not establish a causal effect of diet on probability of recover.

### END SOLUTION

## Question 2

Answer TRUE/FALSE to the following statement:

An *additive* multiple linear regression with one continuous input variable and one categorical input variable with four categories has 4 *slopes* to be estimated.

### BEGIN SOLUTION

FALSE, the additive model has a unique slope shared by all the levels of the categorical variable. We assume that the association between the response and the continuous covariate is the *same* for all levels of the categorical variable.

### END SOLUTION

## Question 2

A data set called `Carseats` containing the number of sales of child car seats in 400 different stores has the following columns:

- `Sales`:  the number of child car seats sold in a store;
- `Price`: the price of car seats in a store;
- `Urban`:  indicates whether the store is in an urban or a rural zone.
- `US`: indicates whether the store is in the US or not.

We start by fitting the following model and obtained the following summary output:

In [4]:
library(ISLR)
summary(lm(formula = Sales ~ Price + Urban + US, data = Carseats))


Call:
lm(formula = Sales ~ Price + Urban + US, data = Carseats)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.9206 -1.6220 -0.0564  1.5786  7.0581 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.043469   0.651012  20.036  < 2e-16 ***
Price       -0.054459   0.005242 -10.389  < 2e-16 ***
UrbanYes    -0.021916   0.271650  -0.081    0.936    
USYes        1.200573   0.259042   4.635 4.86e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.472 on 396 degrees of freedom
Multiple R-squared:  0.2393,	Adjusted R-squared:  0.2335 
F-statistic: 41.52 on 3 and 396 DF,  p-value: < 2.2e-16


**Is the statement below true or false?**

Based on the model used, it was assumed that the association between price and sales differs by country.

### BEGIN SOLUTION

FALSE: an additive model assumes that the association between price and sales is the same in US and non-US countries. The model has a unique slope of -0.0544

### END SOLUTION

## Question 3

Trying to understand the profile of students who are accepted into Medical School, the following data were collected (among other variables not used in this analysis):

`Acceptance`: accepted (1) into medical school or denied (0)

`GPA`: overall GPA

`MCAT`: MCAT total score

`Apps`:  number of schools applied to

The data is available in a data frame called `medgpa`, which we use to fit a model in R.  

**Which of the following codes would you use to build a strong predictive model of the response Acceptance?**

**A**: lm(Acceptance ~ ., data = medgpa)

**B**: glm(Acceptance ~ ., family = binomial, data = medgpa)

**C**: glm(Acceptance ~ ., family = poisson, data = medgpa)

**D**: lm(GPA ~ ., data = medgpa)

### BEGIN SOLUTION

**B**: glm(Acceptance ~ ., family = binomial, data = mdgpa)

The response is binary (accepted or denied). We fit a Logistic regression using the R function `glm()` and the argument `family = binomial`
### END SOLUTION

### Question 4

Following the previous question, and based on the results shown in the table below, **which is a correct interpretation of the coefficient MCAT?** (only one is correct)

Based on the (raw) estimated coefficients (`exponentiate = FALSE` in `tidy()`) displayed in `medgpa_table`

**Which is a correct interpretation of the coefficient MCAT?** (only one is correct)

In [11]:
medgpa_table <- data.frame(
  Variable = c("(Intercept)", "GPA", "MCAT", "Apps"),
  Estimate = c(-23.68942, 4.86062, 0.17287, 0.04379),
  Std_Error = c(7.02387, 1.69441, 0.10537, 0.07617),
  z_value = c(-3.373, 2.869, 1.641, 0.575),
  Pr_greater_than_z = c(0.000744, 0.004123, 0.100867, 0.565412)
)
medgpa_table

Variable    Estimate  Std_Error z_value Pr_greater_than_z
1 (Intercept) -23.68942 7.02387   -3.373  0.000744         
2 GPA           4.86062 1.69441    2.869  0.004123         
3 MCAT          0.17287 0.10537    1.641  0.100867         
4 Apps          0.04379 0.07617    0.575  0.565412

**A**: Keeping GPA and Apps constant (at any value), for every additional point in MCAT we expect the odds of being accepted in med school to increase by a factor of $e^0.17=1.185$ , i.e., by approximately 18.5%

**B**: Keeping GPA and Apps constant (at any value), for every additional point in MCAT we expect the odds of being accepted in med school to increase by approximately $17\%$

**C**: The relation between `MCAT` and `Acceptance` depends on the GPA of the student.

**D**: Keeping `GPA` and `Apps` constant (at any value), the odds of being accepted in med school are approximately 18.5% higher ($e^0.17=1.185$) in students with high MCAT score compared to those with low score.

### BEGIN SOLUTION

**A**

B: is incorrect since 0.17 is the coefficient for a model of the log-odds, not the logs

C: is incorrect the model is additive. We assume that the relation between `MCAT` and `Acceptance` depends NOT on the GPA of the student.

D: is incorrect since MCAT is a continuous variable

### END SOLUTION

**Question 5**

Propose a question that can be answered using the `medgpa` dataset described in the previous Medical School question and a Poisson regression.

a) Explain why a Poisson regression is needed for your question.

b) Justify the choice of variables and model you make.

c) Write a code line to fit the proposed model.

### BEGIN SOLUTION

Is there an association between the number of applications (`Apps`) sent by a student their grades (e.g., `GPA`  and `MCAT`)?

Since the response `Apps` is a discrete variable with counts (number of applications) a Poisson regression can be used to model the mean number of applications as a function of grades.

`glm(Apps ~ GPA + MCAT, family = "poisson", data = medgpa)`

### END SOLUTION